src/main/java/com/knighttodo/knighttodo/service/impl/RoutineServiceImpl.java

```diff
@@ -38,16 +44,24 @@ public RoutineVO findById(String routineId) {
     }
 
     @Override
-    public RoutineVO updateRoutine(String routineId, RoutineVO changedRoutineVO) {
+    public RoutineVO updateRoutine(String blockId, String routineId, RoutineVO changedRoutineVO) {
         RoutineVO routineVO = findById(routineId);
+        routineVO.setTodoBlock(todoBlockService.findById(blockId));
         routineVO.setName(changedRoutineVO.getName());
+        routineVO.setTemplateId(routineId);
         routineVO.setHardness(changedRoutineVO.getHardness());
         routineVO.setScariness(changedRoutineVO.getScariness());
+        routineVO.setReady(changedRoutineVO.isReady());
         return routineGateway.save(routineVO);
     }
 
     @Override
     public void deleteById(String routineId) {
         routineGateway.deleteById(routineId);
     }
+
+    @Override
+    public List<RoutineVO> findAllPatterns() {
+        return routineGateway.findAllTemplates();
+    }
```

Comment:
Why did you decide to not name it same as gateway method? I think it can be better to use synonyms as few times as possible

Reply:
Oh, yes, I forgot to change it. Actually the aroginal word should be Template, but somehow I wrote patterns..